lib/ansible/modules/system/java_cert.py

```diff
@@ -0,0 +1,282 @@
+        keystore_path as alias '''
+    import_cmd = ("keytool -importcert -noprompt -keystore '%s' "
+                  "-storepass '%s' -file '%s' -alias '%s'")%(keystore_path,
+                                                             keystore_pass,
+                                                             path, alias)
+
+    if module.check_mode:
+        module.exit_json(changed=True)
+
+    # Use local certificate from local path and import it to a java keystore
+    (import_rc, import_out, import_err) = module.run_command(import_cmd,
+                                                             check_rc=False)
+
+    diff = {'before': '\n', 'after': '%s\n'%alias}
+    if import_rc == 0:
+        return module.exit_json(changed=True, msg=import_out,
+                                rc=import_rc, cmd=import_cmd, stdout=import_out,
+                                error=import_err, diff=diff)
+    else:
+        return module.fail_json(msg=import_out, rc=import_rc, cmd=import_cmd)
+
+def delete_cert(module, keystore_path, keystore_pass, alias):
+    ''' Delete cerificate identified with alias from keystore on keystore_path '''
+    del_cmd = ("keytool -delete -keystore '%s' -storepass '%s' "
+               "-alias '%s'")%(keystore_path, keystore_pass, alias)
+
+    if module.check_mode:
+        module.exit_json(changed=True)
+
+    # Delete SSL certificate from keystore
+    (del_rc, del_out, del_err) = module.run_command(del_cmd, check_rc=True)
+
+    diff = {'before': '%s\n'%alias, 'after': None}
+
+    return module.exit_json(changed=True, msg=del_out,
+                            rc=del_rc, cmd=del_cmd, stdout=del_out,
+                            error=del_err, diff=diff)
+
+def test_keytool(module):
+    ''' Test if keytool is actuall executable or not '''
+    test_cmd = "keytool"
+
+    module.run_command(test_cmd, check_rc=True)
+
+def test_keystore(module, keystore_path):
+    ''' Check if we can access keystore as file or not '''
+    if keystore_path is None:
+        keystore_path = ''
+
+    if not os.path.exists(keystore_path) and not os.path.isfile(keystore_path):
+        ## Keystore doesn't exist we want to create it
+        return module.fail_json(changed=False,
+                                msg="Module require existing keystore at keystore_path '%s'"
+                                %(keystore_path))
+
+def main():
+    argument_spec = dict(
+        cert_url=dict(required=False, type='str'),
+        cert_path=dict(required=False, type='str'),
+        cert_alias=dict(required=False, type='str'),
+        cert_port=dict(required=False, default='443', type='int'),
+        keystore_path=dict(required=False, type='str'),
+        keystore_pass=dict(required=False, default='changeit', type='str'),
+        keystore_create=dict(required=False, default=False, type='bool'),
+        state=dict(required=False, default='present',
+                   choices=['present', 'absent'])
```

Comment:
Here also if a parameter is not required, we do not add `required=False`, we only add `required=True` when it is required.